From 8f5aac08a11a60c7fe21437c86c01df1367a0ba2 Mon Sep 17 00:00:00 2001
From: hanchao <hanchao47@huawei.com>
Date: Tue, 15 Nov 2022 00:12:14 +0800
Subject: [PATCH] rubik: fix Dockerfile build failed

---
 Dockerfile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Dockerfile b/Dockerfile
index 51f0e4c..78c7702 100644
--- a/Dockerfile
+++ b/Dockerfile
@@ -1,3 +1,3 @@
 FROM scratch
-COPY ./rubik /rubik
+COPY ./build/rubik /rubik
 ENTRYPOINT ["/rubik"]
-- 
2.30.2

